2.4/root/usr/share/container-scripts/mongodb/common.sh

```diff
@@ -117,23 +117,60 @@ function mongo_initiate() {
 
 # get the address of the current primary member
 function mongo_primary_member_addr() {
-  local current_endpoints=$(endpoints)
-  local mongo_node="$(echo "${current_endpoints}" | grep -v "$(container_addr)" | head -1):${CONTAINER_PORT}"
-  echo -n $(mongo admin -u admin -p "${MONGODB_ADMIN_PASSWORD}" --host ${mongo_node} --quiet --eval "print(rs.isMaster().primary);")
+  endpoints | grep -v "$(container_addr)" | while read mongo_node; do
+    cmd_output="$(mongo admin -u admin -p "$MONGODB_ADMIN_PASSWORD" --host "$mongo_node:$CONTAINER_PORT" --eval 'print(rs.isMaster().primary)' --quiet || true)"
+
+    # Trying to find IP:PORT in output and filter out error message because mongo prints it to stdout
+    ip_and_port_regexp='[0-9]\+\.[0-9]\+\.[0-9]\+\.[0-9]\+:[0-9]\+'
+    if addr="$(echo "$cmd_output" | grep -x "$ip_and_port_regexp")"; then
+      echo "$addr"
+      exit 0
+    fi
+
+    echo >&2 "Cannot get address of primary from $mongo_node node: $cmd_output"
```

Comment:
"Bash, every day a new lesson -- guaranteed" :sunrise: 

```console
$ cat assignment.sh 
#!/bin/bash

foo="$(echo "stdout" && echo >&2 "stderr")"

echo "The value of foo is '${foo}'"
$ ./assignment.sh 
stderr
The value of foo is 'stdout'
```

@php-coder when you make an assignment with a subshell, only the `stdout` becomes the value of the variable.
So if the error from `mongo` goes to `stderr`, there is no point in grepping the value stored in `cmd_output`, just like in this particular line it seems that `cmd_output` will always be empty when you try to print it.

Or maybe I'm wrong?

I really miss being able to write unit tests for these functions.